tools/jnimarshalmethod-gen/TypeMover.cs

```diff
@@ -48,12 +48,20 @@ public void Move ()
 
 		Dictionary<string, MethodReference> newHelperMethods;
 
+		bool TypeIsEmptyOrHasOnlyDefaultConstructor (TypeDefinition type)
+		{
+			return !type.HasMethods || (type.Methods.Count == 1 && type.Methods [0].IsConstructor);
```

Comment:
Should we worry/care about static constructors?

Should we only check types matching a certain "well-known" naming convention?

Reply:
We generate these types, so we know their structure. We either generate type with marshaling methods or these "empty" ones, just to have the same the class hierarchy in the *-JniMarshalMethod assembly.